src/derive/curve.rs

```diff
@@ -249,43 +297,45 @@ macro_rules! new_curve_impl {
 
                             x.and_then(|x| {
                                 y.and_then(|y| {
-                                    // Create a point representing this value
+                                    // The point is the identity iff the x and y coordinates are zero
+                                    // and the identity flag is set, in the fomrats where such flag is present.
+                                    let is_ident = infinity_flag_set & x.is_zero() & y.is_zero();
```

Comment:
Let say I have a point that is **not** the identity.  And I encode it as uncompressed with `infinity_flag_set = 1`.  The serialization is incorrect.
Then in the decoding the `is_ident` will be false because `x` or `y` is different than 0.
This function will return the point `(x, y)`.
`from_uncompressed` calls this function and will check if `(x, y)` are in the curve, which they are, so it will return successfully.

So we have a malformed uncompressed point (it has the identity flag but it's not the identity), and it loads without errors via `from_uncompressed`.  I would assume `from_uncompressed` should reject invalid formed encodings?